src/vcpkg/binarycaching.cpp

```diff
@@ -1120,6 +1120,186 @@ namespace
         std::vector<std::string> m_read_prefixes;
         std::vector<std::string> m_write_prefixes;
     };
+
+    bool awscli_stat(const std::string& url)
+    {
+        Command cmd;
+        cmd.string_arg("aws").string_arg("s3").string_arg("ls").string_arg(url);
+        const auto res = cmd_execute(cmd);
+        return res == 0;
+    }
+
+    bool awscli_upload_file(const std::string& aws_object, const Path& archive)
+    {
+        Command cmd;
+        cmd.string_arg("aws").string_arg("s3").string_arg("cp").path_arg(archive).string_arg(aws_object);
+        const auto out = cmd_execute_and_capture_output(cmd);
+        if (out.exit_code == 0)
+        {
+            return true;
+        }
+
+        print2(Color::warning, "aws failed to upload with exit code: ", out.exit_code, '\n', out.output);
+        return false;
+    }
+
+    bool awscli_download_file(const std::string& aws_object, const Path& archive)
+    {
+        Command cmd;
+        cmd.string_arg("aws").string_arg("s3").string_arg("cp").string_arg(aws_object).path_arg(archive);
+        const auto out = cmd_execute_and_capture_output(cmd);
+        if (out.exit_code == 0)
+        {
+            return true;
+        }
+
+        print2(Color::warning, "aws failed to download with exit code: ", out.exit_code, '\n', out.output);
```

Comment:
This should be using the new localized output; at the top of the file:

```cxx
DECLARE_AND_REGISTER_MESSAGE(AwsFailedToDownload, (msg::value, msg::output), "aws failed to download with exit code: {value}\n{output}");
```

(then add a new `DECLARE_MSG_ARG` in messages.h for `output`)

Then, replace this with:

```suggestion
        msg::println(Color::warning, msgAwsFailedToDownload, msg::value = out.exit_code, msg::output = out.output);
```